Sources/System/InternetProtocol.swift

```diff
@@ -0,0 +1,208 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+/// Internet Protocol Address
+@frozen
+public enum IPAddress: Equatable, Hashable, Codable {
+    
+    /// IPv4
+    case v4(IPv4Address)
+    
+    /// IPv6
+    case v6(IPv6Address)
+}
+
+public extension IPAddress {
+    
+    @_alwaysEmitIntoClient
+    func withUnsafeBytes<Result>(_ body: ((UnsafeRawBufferPointer) -> (Result))) -> Result {
+        switch self {
+        case let .v4(address):
+            return address.withUnsafeBytes(body)
+        case let .v6(address):
+            return address.withUnsafeBytes(body)
+        }
+    }
+}
+
+extension IPAddress: RawRepresentable {
+    
+    public init?(rawValue: String) {
+        
+        if let address = IPv4Address(rawValue: rawValue) {
+            self = .v4(address)
+        } else if let address = IPv6Address(rawValue: rawValue) {
+            self = .v6(address)
+        } else {
+            return nil
+        }
+    }
+    
+    public var rawValue: String {
+        switch self {
+        case let .v4(address): return address.rawValue
+        case let .v6(address): return address.rawValue
+        }
+    }
+}
+
+extension IPAddress: CustomStringConvertible, CustomDebugStringConvertible {
+    
+    public var description: String {
+        return rawValue
+    }
+    
+    public var debugDescription: String {
+        return description
+    }
+}
+
+/// IPv4 Socket Address
+@frozen
+public struct IPv4Address: Equatable, Hashable, Codable {
```

Comment:
I don't inherently disagree with that, but we either have to have a useful way to project them to a number or to bytes, otherwise getting them _as_ a sequence of bytes is a pain in the neck. There are lots of reasons to want to be able to do that, most notably when working as network protocols, and having a safe way to do it is useful. Given that projecting an IPv6 address as an integer is not possible in Swift without swift-numerics (an unreasonable dependency for this project), I'm inclined to want to still offer it.